src/radical/pilot/agent/scheduler/continuous_fifo.py

```diff
@@ -0,0 +1,51 @@
+
+__copyright__ = "Copyright 2013-2016, http://radical.rutgers.edu"
+__license__   = "MIT"
+
+
+from .continuous import Continuous
+
+# ------------------------------------------------------------------------------
+#
+# This is a simle extension of the Continuous scheduler which makes RP behave
+# like a FiFo: it will only really attempt to schedule units if all older units
+# (units with lower unit ID) have been scheduled.  As such, it relies on unit
+# IDs to be of the format `unit.%d'`.
+#
+class ContinuousFifo(Continuous):
+
+    # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg, session):
+
+        self.slots = None
+        self._last = -1   # nothing has run, yet - next up is unit.000000
+
+        Continuous.__init__(self, cfg, session)
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _try_allocation(self, cu):
+
+        # check if the unit is up for scheduling
+        uid = cu['uid']
+        idx = int(uid.split('.')[1])
+
+        if idx > self._last + 1:
```

Comment:
Question: Given that all units and their units are created in sequence (i.e. unit.000003 is created only after unit.000002), can we change this condition to :

```
if idx == self._last + 1: 
    # yep: lets try for real
    ret = Continuous.try_allocatio(self, cu)

    if ret: 
        # it worked!  Keep the new index
        self._last = idx

    return ret

else:
    # nope - we let this attempt fail
    self._log.debug('defer unit %s [%s]', uid, self._last)
    return False
```

This question/suggestion is only for ease of understanding of code (if my assumption on sequence of uids is always correct).

Reply:
The only difference is the inversion of the `if` test - or am I missing something?  I find both equally easy to read - the whole method is just 10 lines after all, so no preference.  In longer methods I always prefer to have the shorter `if`-branch first, to keep the `else` clause visually close.  And I usually also try to bail-out (return) from a method as early as possible, as soon as that information exists - that avoids too many `else`-clauses.  But, as said, no preference in this case - feel free to adjust either way.

Reply:
I agree it is small. I'll add it in a comment then. My concern is not really about the if and else order, it was more about testing '==' instead of '>'. I think using '!=' in the if condition both our suggestions. Let me modify that then.

Reply:
Ah, I see - but `!=` would in principle also test for smaller indexes, which by definition cannot happen, right?  Wouldn't that increase confusion for readers? :-)

Reply:
That's what I added in the comment above the if condition :)

The '>' condition seems redundant if you know that uids are in order of unit creation. The '!=' seems incomplete (since it might test for small indexes) if you don't know that the uids are in order of unit creation. My preference is for the latter with a note :)